Declutter README (#2232)
nf-core/tools: nf_core/pipeline-template/README.md

## Introduction

**{{ name }}** is a bioinformatics pipeline that ...

<!-- TODO nf-core:

>    Complete this sentence with a 2-3 sentence summary of what types of data the pipeline ingests, a breif overview of the

Reviewer: ```suggestion
   Complete this sentence with a 2-3 sentence summary of what types of data the pipeline ingests, a brief overview of the
```

   major pipeline sections and the types of output it produces. You're giving an overview to someone new
   to nf-core here, in 15-20 seconds. For an example, see https://github.com/nf-core/rnaseq/blob/master/README.md#introduction
-->

<!-- TODO nf-core: Include a figure that guides the user through the major workflow steps. Many nf-core
     workflows use the "tube map" design for that. See https://nf-co.re/docs/contributing/design_guidelines#examples for examples.   -->
<!-- TODO nf-core: Fill in short bullet-pointed list of the default steps in the pipeline -->

1. Read QC ([`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/))
2. Present QC for raw reads ([`MultiQC`](http://multiqc.info/))

## Usage

> **Note**

> > If you are new to nextflow and nf-core, please refer to [this page](https://nf-co.re/docs/usage/installation) on how
> > to set-up nextflow. Make sure to [test your setup](https://nf-co.re/docs/usage/introduction#how-to-run-a-pipeline)

Reviewer: ```suggestion
> If you are new to Nextflow and nf-core, please refer to [this page](https://nf-co.re/docs/usage/installation) on how
> to set-up Nextflow. Make sure to [test your setup](https://nf-co.re/docs/usage/introduction#how-to-run-a-pipeline)
```

> with `-profile test` before running the workflow on actual data.

<!-- TODO nf-core: Describe the minimum required steps to execute the pipeline, e.g. how to prepare samplesheets.
     Explain what rows and columns represent. For instance (please edit as appropriate):

> First, you need to prepare a samplesheet with your input data that looks as follows:

Reviewer: ```suggestion
First, prepare a samplesheet with your input data that looks as follows:
```

`samplesheet.csv`:

```csv
sample,fastq_1,fastq_2
CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz
```

Each row represents a fastq file (single-end) or a pair of fastq files (paired end).

-->

Now, you can run the pipeline using:

<!-- TODO nf-core: update the following command to include all required parameters for a minimal example -->

```bash
nextflow run {{ name }} \
   -profile <docker/singularity/.../institute> \
   --input samplesheet.csv \
   --outdir <OUTDIR>
```

> **Warning:**
> Please provide pipeline parameters via the CLI or Nextflow `-params-file` option. Custom config files including those
> provided by the `-c` Nextflow option can be used to provide any configuration _**except for parameters**_;
> see [docs](https://nf-co.re/usage/configuration#custom-configuration-files).

{% if branded -%}

For more details, please refer to the [usage documentation](https://nf-co.re/{{ short_name }}/usage) and the [parameter documentation](https://nf-co.re/{{ short_name }}/parameters).

## Pipeline output

> To see the the results of a test run with a full size dataset refer to the [results](https://nf-co.re/{{ short_name }}/results) section of the documentation.

Reviewer: ```suggestion
To see the the results of a test run with a full size dataset refer to the [results](https://nf-co.re/{{ short_name }}/results) tab on the nf-core website pipeline page.
```

## Introduction

<!-- TODO nf-core: Write a 2-3 sentence summary of what data the pipeline is for and what it does, e.g.

>    > **nf-core/rnaseq** is a bioinformatics pipeline that can be used to analyse RNA sequencing data obtained from

Reviewer: ```suggestion
   > **{{name}}** is a bioinformatics pipeline that can be used to analyse RNA sequencing data obtained from
```

Reviewer: I think we should avoid anything about the rnaseq pipeline, rather keep this generic.

>    > **nf-core/rnaseq** is a bioinformatics pipeline that can be used to analyse RNA sequencing data obtained from
>    > organisms with a reference genome and annotation. It takes a samplesheet and FASTQ files as input,
>    > performs quality control (QC), trimming and (pseudo-)alignment, and produces a gene expression matrix and extensive QC report.

Reviewer: ```suggestion
   > **{{ name }}** is a bioinformatics pipeline that can.. Please complete these sentences with a description
   > of the types of data that the pipeline ingests, a brief overview of the major pipeline sections and the types of
   > outputs that it produces. You're giving an overview to someone new to nf-core here, in 15-20 seconds.
```

Author: Personally, I find an example much more instructive than the text (this is why stackoverflow took off in the first place). But I don't mind your version too much, or we could have both.

Reviewer: I mostly just found it a bit jarring as I read through. Maybe we could put in a couple of links to other pipelines as examples?

-->

> **{{ name }}** is a bioinformatics pipeline for {{ description }}.

Reviewer: Seems to be a duplicate of the blockquote above?

<!-- TODO nf-core: Include a figure that guides the user through the major workflow steps. Many nf-core

>      workflows use the "tube map" design for that.  -->

Reviewer 2: ```suggestion
     workflows use the "tube map" design for that. See https://nf-co.re/docs/contributing/design_guidelines#examples for examples.   -->
```

<!-- TODO nf-core: Fill in short bullet-pointed list of the default steps in the pipeline -->

1. Read QC ([`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/))
2. Present QC for raw reads ([`MultiQC`](http://multiqc.info/))

## Usage

> {% if branded -%}

Reviewer 3: I would show this note also for unbranded pipelines. Maybe only remove the `nf-core` name?

Author: I agree it could be nice to have this, it still points to the nf-core website, though :thinking: 
What do others think?

Reviewer: I think links to general docs are ok. Branding is more about identifying the pipeline as an nf-core pipeline, I don't get the impression that this paragraph is doing that.

> **Note**
> If you are new to nextflow and nf-core, please refer to [this page](https://nf-co.re/docs/usage/installation) on how
> to set-up nextflow. Make sure to [test your setup](https://nf-co.re/docs/usage/introduction#how-to-run-a-pipeline)
> with `-profile test` before running the workflow on actual data.

> {% endif -%}

Author: ```suggestion
```

<!-- TODO nf-core: Describe the minimum required steps to execute the pipeline, e.g. how to prepare samplesheets.

>      Explain what rows and columns represent.
>  -->

Reviewer: ```suggestion

For example: (please edit as appropriate)

```

> First, you need to prepare a samplesheet with your input data that looks as follows:

Reviewer: I wonder if this example content should be within the comment block. It won't be appropriate for many pipelines.

> **samplesheet.csv**:

Reviewer: ```suggestion
`samplesheet.csv`:
```

```csv
sample,fastq_1,fastq_2
CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz
```

> Each row represents a fastq file (single-end) or a pair of fastq files (paired end).

Reviewer: ```suggestion
Each row represents a fastq file (single-end) or a pair of fastq files (paired end).

-->
```

> **Warning**
> Please provide pipeline parameters via the CLI or Nextflow `-params-file` option. Custom config files including those
> provided by the `-c` Nextflow option can be used to provide any configuration **except for parameters**;

> > see [docs](https://nf-co.re/usage/configuration#custom-configuration-files).

Reviewer 2: I would move this below the actual command itself

```bash

> nextflow run {{ name }} \
>    --input samplesheet.csv \
>    -profile <docker/singularity/.../institute>

Reviewer: Any reason to drop outdir and genome? Outdir is typically a required param in basically all pipelines, so I think it should be there.

Might also be good to put core options such as `-profile` first so as not to mix numbers of hyphens.

```suggestion
nextflow run {{ name }} \
   -profile <docker/singularity/.../institute> \
   --input samplesheet.csv \
   --genome GRCh37
   --outdir <OUTDIR>
```

Reviewer 2: Agreed, genome is actually only specific to a small subset of pipelines (many will have `fasta` instead as they need to support non iGenomes referecnes; or have no reference at all)

Reviewer: ok sure 👍🏻

Reviewer: ```suggestion
nextflow run {{ name }} \
   -profile <docker/singularity/.../institute> \
   --input samplesheet.csv \
   --outdir <OUTDIR>
```

```

> > **Warning**

Reviewer: Need a `:` for the nice warning block formatting on GitHub.

```suggestion
> **Warning:**
```

> Please provide pipeline parameters via the CLI or Nextflow `-params-file` option. Custom config files including those
> provided by the `-c` Nextflow option can be used to provide any configuration **except for parameters**;
> see [docs](https://nf-co.re/usage/configuration#custom-configuration-files).

{% if branded -%}

For more details, please refer to the [usage documentation](https://nf-co.re/{{ short_name }}/usage) and the [parameter documentation](https://nf-co.re/{{ short_name }}/parameters).

{% endif -%}

{% if branded -%}

## Pipeline output

> [results](https://nf-co.re/{{ short_name }}/results) section of the documentation.

Reviewer 3: ```suggestion
To see the the results of a test run with a full size dataset refer to the [results](https://nf-co.re/{{ short_name }}/results) section of the documentation.
```

Author: good catch, thank you! Seems something got lost in the last commit :/

> For more details, please refer to the [output documentation](https://nf-co.re/{{ short_name }}/output).

Reviewer 2: ```suggestion
You can see an example of the output of a realistic run on on the [results documetnation](https://nf-co.re/{{ short_name }}/results) page. For more specific details about the output files and reports, please refer to the [output documentation](https://nf-co.re/{{ short_name }}/output).
```

Link not tested

Reviewer 2: Yes it does exist, albeit only has files inside after the first release of the pipeline. But until then there is a big warning saying it's not yet released so I think that should be fine.

E.g.

https://nf-co.re/spinningjenny/results#spinningjenny/
